src/data/equiv/fintype.lean

```diff
@@ -75,3 +75,43 @@ by rwa [equiv.perm.via_embedding, equiv.perm.extend_domain_apply_not_subtype]
 @[simp] lemma equiv.perm.via_embedding_sign [decidable_eq α] [fintype β] :
   equiv.perm.sign (e.via_embedding f) = equiv.perm.sign e :=
 by simp [equiv.perm.via_embedding]
+
+namespace equiv
+variables {p q : α → Prop} [decidable_pred p] [decidable_pred q]
+
+/-- If `e` is an equivalence between two subtypes of a fintype `α`, `e.to_compl`
+is an equivalence between the complement of those subtypes. -/
+noncomputable def to_compl (e : {x // p x} ≃ {x // q x}) : {x // ¬ p x} ≃ {x // ¬ q x} :=
+classical.choice (fintype.card_eq.mp (fintype.card_compl_eq_card_compl (fintype.card_congr e)))
+
+/-- If `e` is an equivalence between two subtypes of a fintype `α`, `e.extend_subtype`
+is a permutation of `α` acting like `e` on the subtypes and doing something arbitrary outside. -/
+noncomputable def extend_subtype (e : {x // p x} ≃ {x // q x}) : perm α :=
+(equiv.sum_compl p).symm.trans ((equiv.sum_congr e e.to_compl).trans
+  (equiv.sum_compl q))
```

Comment:
I think it might be worth adding a more general version, and building the non-computable one from it:
```suggestion
/-- `equiv.perm.subtype_congr
def subtype_congr (e : {x // p x} ≃ {x // q x}) (f : {x // ¬p x} ≃ {x // ¬q x}) : perm α :=
(equiv.sum_compl p).symm.trans ((equiv.sum_congr e f).trans
  (equiv.sum_compl q))
  
abbreviation extend_subtype (e : {x // p x} ≃ {x // q x}) (f : {x // ¬p x} ≃ {x // ¬q x}) : perm α :=
subtype_congr e e.to_compl
```